backend/src/main/java/com/bikes/backend/config/SecurityConfig.java

```diff
@@ -0,0 +1,55 @@
+package com.bikes.backend.config;
+
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.crypto.argon2.Argon2PasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.logout.HttpStatusReturningLogoutSuccessHandler;
+import org.springframework.security.web.csrf.CookieCsrfTokenRepository;
+import org.springframework.security.web.csrf.CsrfTokenRequestAttributeHandler;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig {
+
+	@Bean
+	public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+
+		CsrfTokenRequestAttributeHandler requestHandler = new CsrfTokenRequestAttributeHandler();
+		requestHandler.setCsrfRequestAttributeName(null);
+
+		return http
+				.csrf(csrf -> csrf.csrfTokenRepository(CookieCsrfTokenRepository.withHttpOnlyFalse()).csrfTokenRequestHandler(requestHandler))
+				.httpBasic()
+				.authenticationEntryPoint((request, response, authException) -> response.sendError(HttpStatus.UNAUTHORIZED.value(), HttpStatus.UNAUTHORIZED.getReasonPhrase()))
+				.and()
+				.sessionManagement(config ->
+						config.sessionCreationPolicy(SessionCreationPolicy.ALWAYS))
+				.authorizeHttpRequests()
+				.requestMatchers(HttpMethod.GET, "/api/csrf/").permitAll() // Everyone may get an CSRF TOKEN - controller tested
+				.requestMatchers(HttpMethod.POST, "/api/users/").permitAll() // Everyone may sign up - controller tested
+				.requestMatchers(HttpMethod.POST, "/api/users/login").permitAll() // Everyone may log in
+				.requestMatchers("/api/users/admin").hasRole("ADMIN") // controller tested
+				.anyRequest().authenticated()
```

Comment:
It looks like you are blocking your frontend with this line, is this possible? We suggest:
```suggestion
				.requestMatchers("/api/**").authenticated()
                                .anyRequest().permitAll()
```